### PR TITLE
[ACA-2651] Viewer - toolbar disappears after marking / removing a specific file as favourite

### PR DESCRIPTION
**Pull request by pionnegru**
## PR Checklist
Please check if your PR fulfills the following requirements:

```
- [x] The commit message follows our guidelines: https://github.com/Alfresco/alfresco-content-app/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
```

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application / Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: [ACA-2651](https://issues.alfresco.com/jira/browse/ACA-2651)

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/components/toolbar/toggle-favorite/toggle-favorite.component.ts [hunk 1]
```diff
@@ -54,14 +55,28 @@ import {
...
 
+  ngOnInit() {
+    this.reloadOnRoutes = this.extensionService.runExpression(
+      `$( ${this.data} )`
+    );
+  }
+
   onToggleEvent() {
-    this.store.dispatch(new ReloadDocumentListAction());
+    if (this.reloadOnRoutes.includes(this.router.url)) {
```
Comment (suzanadirla) [inline src/app/components/toolbar/toggle-favorite/toggle-favorite.component.ts]: please make sure that this.reloadOnRoutes is defined

File: src/app/components/layout/app-layout/app-layout.component.ts [hunk 2]
```diff
@@ -140,15 +141,14 @@ export class AppLayoutComponent implements OnInit, OnDestroy {
...
-            !event.url.startsWith('/search;')
-          );
+          return event instanceof NavigationStart;
         }),
         takeUntil(this.onDestroy$)
       )
-      .subscribe(() => this.store.dispatch(new SetSelectedNodesAction([])));
+      .subscribe(() => {
+        this.store.dispatch(new SetSelectedNodesAction());
+        this.store.dispatch(new ReloadDocumentListAction());
```
Comment (suzanadirla) [inline src/app/components/layout/app-layout/app-layout.component.ts]: I have realized that this reload should not be done on each route change. Maybe we could trigger it for '/search' only on toggle-favorite in a similar way as we do on '/favorites' and '/favorite/libraries'

Comment (pionnegru) [reply src/app/components/layout/app-layout/app-layout.component.ts]: yes, I know

File: src/app/components/search/search-results-row/search-results-row.component.ts [hunk 3]
```diff
@@ -121,11 +121,13 @@ export class SearchResultsRowComponent implements OnInit, OnDestroy {
     return this.node.entry.isFile;
   }
 
-  showPreview() {
+  showPreview(event) {
```
Comment (DenysVuika) [inline src/app/components/search/search-results-row/search-results-row.component.ts]: misses type information

File: src/app/components/search/search-results-row/search-results-row.component.ts [hunk 4]
```diff
@@ -121,11 +121,13 @@ export class SearchResultsRowComponent implements OnInit, OnDestroy {
...
+    event.stopPropagation();
     this.store.dispatch(new ViewFileAction(this.node));
   }
 
-  navigate() {
+  navigate(event) {
```
Comment (DenysVuika) [inline src/app/components/search/search-results-row/search-results-row.component.ts]: event misses type information

File: src/app/components/layout/app-layout/app-layout.component.ts (hunk 2 above)
Comment (pionnegru) [reply src/app/components/layout/app-layout/app-layout.component.ts]: as agreed, we will raise a separate ticket

**Pull request merged**